1.1 Method Missing/exercise4.rb

```diff
@@ -2,34 +2,34 @@
 #      us on what they told you.
 
 class MethodCall
-    def initialize(sym, args)
-      @sym = sym
-      @args = args
-    end
-    
-    def sym
-      @sym
-    end
-    
-    def args
-      @args
-    end
-    
-    def ==(other_call)
-      @sym == other_call.sym && @args == other_call.args
-    end
+  def initialize(sym, args)
+    @sym = sym
+    @args = args
   end
   
-  class Spy
-    def initialize
-      @method_calls = []
-    end
-    
-    # Write your method_missing here
+  def sym
+    @sym
+  end
+  
+  def args
+    @args
+  end
+  
+  def ==(other_call)
+    @sym == other_call.sym && @args == other_call.args
+  end
+end
+
+class Spy
+  def initialize
+    @method_calls = []
+  end
   
-    def method_called?(sym, *args)
-      # Your superiors will call this method to ask you if you've seen
-      # a particular method call. Simply answer them by returning true
-      # or false.
-    end
+  def method_missing sym, *args, &block
+    @method_calls.push(MethodCall.new(sym, args))
+  end
+
+  def method_called?(sym, *args)
+    return @method_calls.include?(MethodCall.new(sym, args))
```

Comment:
it's not necessary. Method returns last value by default

Reply:
fixed! but it's better to see written `return ` in code, IMHO